### PR TITLE
Update interfaces

### LINKED ISSUE #228: Switch to Ruby 3.0
Can we update Ruby to 3.0? Ruby 2.4 is already EOL, and so is 2.5. https://www.ruby-lang.org/en/downloads/

RubyInstaller has the both 2.7 and 3.0 versions, so it'll be nice if we switch to Ruby 3.0, but 2.7 is also a good option for now.

### LINKED ISSUE #230: Switch to Python 3.10
Switch to python 3.10 plz:

https://www.python.org/downloads/release/python-3100/

### PR DESCRIPTION
**Pull request by k-takata**
* Lua 5.4
* Perl 5.32
* Python3 3.10 (Close #230)
* Ruby 3.0 (Close #228)

### CODE REVIEW

File: appveyor.bat [hunk 1]
```diff
@@ -129,18 +130,22 @@ start /wait downloads\tcl.exe /extract:c:\ActiveTclTemp /exenoui /exenoupdates /
...
 cmd /c start /wait downloads\python3.exe /quiet TargetDir=%PYTHON3_DIR%  Include_pip=0 Include_tcltk=0 Include_test=0 Include_tools=0 AssociateFiles=0 Shortcuts=0 Include_doc=0 Include_launcher=0 InstallLauncherAllUsers=0
 
 :: Ruby
+:: Download RubyInstaller binary
+call :downloadfile %RUBY_URL% downloads\ruby.7z
+7z x downloads\ruby.7z -oC:\ > nul || exit 1
+move C:\rubyinstaller-%RUBY_RELEASE%-%ARCH% %RUBY_DIR% > nul || exit 1
 :: RubyInstaller is built by MinGW, so we cannot use header files from it.
 :: Download the source files and generate config.h for MSVC.
 rem git clone https://github.com/ruby/ruby.git -b %RUBY_BRANCH% --depth 1 -q ../ruby
```
Comment (chrisbra) [inline appveyor.bat:144]: can we remove that comment?

**Comment by chrisbra** [comment]
looks good, thanks

File: appveyor.bat (hunk 1 above)
Comment (k-takata) [reply appveyor.bat:144]: L144 can be removed.
L142-L143 are still relevant.

Comment (k-takata) [reply appveyor.bat:144]: I keep this for now.

Comment (chrisbra) [reply appveyor.bat:144]: sure thanks

**Comment by skywind3000** [comment]
thanks

**Pull request merged**